prov/util/src/util_ns.c

```diff
@@ -312,13 +313,16 @@ static void *util_ns_name_server_func(void *args)
 	if (ret)
 		goto done;
 
-	cleanup_args[0] = (void *)(uintptr_t)listenfd;
+	cleanup_args[0] = (void *)(uintptr_t)&listenfd;
 	cleanup_args[1] = (void *)ns;
 	pthread_cleanup_push(util_ns_name_server_cleanup,
 			     (void *)cleanup_args);
 
 	while (1) {
-		connfd = accept(listenfd, NULL, 0);
+		if (listenfd != INVALID_SOCKET)
+			connfd = accept(listenfd, NULL, 0);
+		else
+			connfd = INVALID_SOCKET;
```

Comment:
It is better to exit the loop if listenfd becomes INVALID_SOCKET,  otherwise the loop will keep going forever burning CPU cycles for no purpose.

Reply:
I agree. Good catch!
Done
Thanks!